app/util/conf/base_config_loader.py

```diff
@@ -139,12 +145,14 @@ def _load_section_from_config_file(self, config, config_filename, section):
             config_parsed = ConfigFile(config_filename).read_config_from_disk()
 
         if section not in config_parsed:
-            raise _InvalidConfigError('The config file {} does not contain a [{}] section'
-                                      .format(config_filename, section))
+            raise InvalidConfigError('The config file {} does not contain a [{}] section'
+                                     .format(config_filename, section))
+
         clusterrunner_config = config_parsed[section]
+        whitelisted_file_keys = self._get_config_file_whitelisted_keys()
         for key in clusterrunner_config:
-            if key not in config_key_validation:
-                raise _InvalidConfigError('The config file contains an invalid key: {}'.format(key))
+            if key not in whitelisted_file_keys:
+                raise InvalidConfigError('The config file contains an invalid key: {}'.format(key))
```

Comment:
I probably agree with that. Feel free to put it in a new enhancement ticket if you wanna. :wink: